packages/plugin-hardhat/src/index.ts

```diff
@@ -110,13 +113,45 @@ subtask(TASK_COMPILE_SOLIDITY_COMPILE, async (args: RunCompilerArgs, hre, runSup
   const { isFullSolcOutput } = await import('./utils/is-full-solc-output');
   if (isFullSolcOutput(output)) {
     const decodeSrc = solcInputOutputDecoder(args.input, output);
-    const validations = validate(output, decodeSrc, args.solcVersion);
+
+    const namespacedInput = makeNamespacedInput(args.input, output);
+    const { output: namespacedOutput } = await runSuper({ ...args, input: namespacedInput });
+    checkNamespacedCompileErrors(namespacedInput, namespacedOutput);
+
+    const validations = validate(output, decodeSrc, args.solcVersion, args.input, namespacedOutput);
     await writeValidations(hre, validations);
   }
 
   return { output, solcBuild };
 });
 
+/**
+ * Checks for compile errors in the modified contracts for namespaced storage.
+ * If errors are found, throws an error with the compile error messages, and logs
+ * the modified contract source code as debug.
```

Comment:
I wonder if we should log the input source code instead, it would probably give us more information to debug, and we can recover the namespaced input from it.

Reply:
The purpose of the debug logging statements here are to allow us to ask the user to provide this info if they are comfortable with it.  If we want the original source code, we could ask for that instead.  In that case, we could even remove the debug messages from this function since the error itself would be thrown anyways.  What do you think?